### ISSUE TITLE
Blueprint errors inconsistant with 0.11 in 0.12.0-rc3

### ISSUE THREAD

**Issue by robgrimball** [balderdashy/sails#3273]
I am not sure if this was intended or not, but it seems like a bad regression if it was intended.  0.11 had a fairly good structure for presenting errors back to the client from blueprint calls, giving you the error code sails had, as well as raw info if you wanted that and so forth which I'm sure we are all familiar with.

0.12.0-rc3 seems to only be returning an error object with a "stack" property and a "message" property. There is information in the message property which I guess could be parsed out to find the same info as before, but it would be a lot nicer if we still had what we had in 0.11.x.

This is using mongo and sails-mongo adapter in case the issue is really related to those rather than sails/waterline, but since all I'm doing to get this change is to change sails from 0.11.2 to 0.12.0-rc3, I thought I'd put the issue here for now until/unless directed elsewhere.

**Comment by sgress454** [comment]
Thanks for the report @robgrimball, we'll take a look.

**Comment by sailsbot** [comment]
Thanks for posting, @robgrimball.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by robgrimball** [comment]
Given this was given a label of being a bug, it seems like it should remain open until fixed?

**Comment by sgress454** [comment]
Yup, totally--Sailsbot is supposed to ignore things labelled as "bug", but I guess it hasn't learned its lesson yet...

**Comment by fernandolguevara** [comment]
@robgrimball check [this](https://github.com/balderdashy/sails/issues/3440)

**Comment by mikermcneil** [comment]
@robgrimball I believe that was fixed sometime in the 1*th of November- should stay open now.  Sorry about that!

**Comment by mikermcneil** [comment]
@robertd @robgrimball  looking into this today, thanks for the ping

**Comment by mikermcneil** [comment]
see https://github.com/balderdashy/sails/pull/3462#issuecomment-169570569

**Comment by robgrimball** [comment]
I've been experiencing some PC issues today, but hopefully they are clearing up and I will try to get this tested out tomorrow (Fri 1/8) to let you know!  Thanks!

**Comment by robgrimball** [comment]
Mike,

I finally got my PC issues resolved enough to test this out.

This fix seems to be very close.  In fact, it is a little odd what I am seeing, but now there is both a _e attribute with the raw error info in it, and an originalError with the raw error info in it, but no "raw" attribute with the raw error info in it like before.  Either _e or originalError are both sufficient for my purposes and I can rewrite my code to use one of those instead of raw, but if you are shooting for complete backward compatibility then getting that information into a "raw" attribute would accomplish it.

Basically, it fixes the problem mostly, and for my purposes if this is the standard going forward.  It will still be a problem for others of course, and unless fixed completely (by putting the info in the "raw" attribute) then probably should be document to ease some hair pulling.

Rob

**Comment by mikermcneil** [comment]
@robgrimball This is definitely not the expected behavior-- wrote up a bit about it here: https://github.com/balderdashy/sails/commit/b8c3813281a041c0b24db381b046fecfa81a14b7#commitcomment-18455430
